module/VuFind/src/VuFind/Db/Service/OaiResumptionService.php

```diff
@@ -68,11 +68,61 @@ public function removeExpired(): void
      *
      * @param string $token The resumption token to retrieve.
      *
-     * @return ?OaiResumptionEntityInterface
+     * @return     ?OaiResumptionEntityInterface
+     * @deprecated Use OaiResumptionService::findWithId
      */
     public function findToken(string $token): ?OaiResumptionEntityInterface
     {
-        return $this->getDbTable('oairesumption')->findToken($token);
+        return $this->findWithId($token);
+    }
+
+    /**
+     * Retrieve a row from the database based on primary key; return null if it
+     * is not found.
+     *
+     * @param string $id Id to use for the search.
+     *
+     * @return ?OaiResumptionEntityInterface
+     */
+    public function findWithId(string $id): ?OaiResumptionEntityInterface
+    {
+        return $this->getDbTable('oairesumption')->findWithId($id);
+    }
+
+    /**
+     * Retrieve a row from the database based on token; return null if it
+     * is not found.
+     *
+     * @param string $token Token used for the search.
+     *
+     * @return ?OaiResumptionEntityInterface
+     */
+    public function findWithToken(string $token): ?OaiResumptionEntityInterface
+    {
+        return $this->getDbTable('oairesumption')->findWithToken($token);
+    }
+
+    /**
+     * Try to find with token first, if not found then try to find with id where the token is null.
+     *
+     * @param string $tokenOrId Token or id
+     *
+     * @return     ?OaiResumptionEntityInterface
+     * @deprecated This function is for BC and can be removed in the future.
```

Comment:
Maybe this should have a `@todo` tag rather than a `@deprecated` tag, since for the moment, we still need it and can't deprecate it. Something like `@todo In future, we should migrate data to prevent null token fields, which will make this method obsolete.`

Might also be worth renaming to something like `findWithTokenOrLegacyIdToken` to make its purpose slightly more clear.

Reply:
oh, didnt remember the todo comment